dali/pipeline/operator/builtin/external_source_test.cc

```diff
@@ -134,8 +134,7 @@ class ExternalSourceTest : public::testing::WithParamInterface<int>,
     auto &tensor_gpu_list = ws.Output<GPUBackend>(0);
     TensorList<CPUBackend> tensor_cpu_list;
     tensor_cpu_list.Copy(tensor_gpu_list, (ws.has_stream() ? ws.stream() : 0));
-    cudaStreamSynchronize(ws.has_stream() ? ws.stream() : 0);
-
+    cudaDeviceSynchronize();
```

Comment:
it might be that this just "fixes" the problem because it affects the timing. I think the CUDA stream synchronize should do the job here. Maybe the problem is somewhere else?

Reply:
The stream usage in the test just falls back to 0 here (I think), as the ws we pass to Outputs is cleared, so we use 0.
The stream 0 should implicitly sync unless we create non-blocking streams or if we create per-thread 0 stream. I wonder if we have some different defaults for x86 and Xavier?

Reply:
The executor appears uses non-blocking streams, it's still weird that this happens only on Xavier.